calculating_n50_assemblies.py

```diff
@@ -0,0 +1,91 @@
+import glob
+import re
+import gzip
+import operator
+import sys
+import os
+
+def main(in_dir = None, out_dir = None):
+    if os.path.exists(in_dir) == False:
+        sys.exit("Input directory or path incorrect. Please check")
+
+    if os.path.exists(out_dir) == False:
+        os.makedirs(out_dir)
+    
+    file1 = open(os.path.join(out_dir+'/summary_concise.txt'),'w')
+    file2 = open(os.path.join(out_dir+'/summary_verbose.txt'),'w')
+
+    #intializing assembly count
+    assembly_count = 0
+
+    #iterating through each assembly file
+    for file in glob.glob(os.path.join(in_dir+'/*.gz')):
+        file2.write ("Opened file {}\n".format(file))
+        assembly_count += 1
+        with gzip.open(file, mode = 'rt') as f:
+
+            #initializing contig length dictionary and counting contigs
+            contig_length_dict = {}
+            contig_count = 0
+
+            #reading assembly file line by line
+            for line in f:
+
+                #capturing assembly name. Although this gets updated with every contig header,
+                #it remains same for every file due to the nature of the regex pattern
+                assembly = re.findall(r'^>([A-Z]+?0\d).+',line)
+                if len(assembly) > 0:
+                    assembly_name = assembly[0] 
+
+                #capturing contig name, initializing contig length to zero, incrementing contig count       
+                x = re.findall(r'^>(.+?)\s', line)
+                if len(x) > 0:
+                    contig_name = x[0]
+                    contig_length = 0
+                    contig_count += 1
+                
+                #if the program doesn't find a contig header, the seqeunce length of every line gets added
+                #and the contig length gets updated within the contig length dictionary
+                else:
+                    contig_length += len(line) - 1
```

Comment:
Instead of doing `len(line) - 1`, it would be better to strip the newline characters from the line. https://docs.python.org/3/library/stdtypes.html?highlight=strip#str.rstrip is a potential option for doing this.

Reply:
len(line.rstrip('\n')